components/dashboard/src/workspaces/start-workspace-modal-context.tsx

```diff
@@ -0,0 +1,37 @@
+/**
+ * Copyright (c) 2022 Gitpod GmbH. All rights reserved.
+ * Licensed under the GNU Affero General Public License (AGPL).
+ * See License-AGPL.txt in the project root for license information.
+ */
+
+import React, { createContext, useEffect, useState } from 'react';
+
+export const StartWorkspaceModalContext = createContext<{
+    isStartWorkspaceModalVisible?: boolean,
+    setIsStartWorkspaceModalVisible: React.Dispatch<boolean>,
+}>({
+    setIsStartWorkspaceModalVisible: () => null,
+});
+
+export const StartWorkspaceModalContextProvider: React.FC = ({ children }) => {
+    const [ isStartWorkspaceModalVisible, setIsStartWorkspaceModalVisible ] = useState<boolean>(false);
+
+    useEffect(() => {
+        const onKeyDown = (event: KeyboardEvent) => {
+            if ((event.metaKey || event.ctrlKey) && event.key === 'o') {
+                event.preventDefault();
+                setIsStartWorkspaceModalVisible(true);
+            }
+        };
+        window.addEventListener('keydown', onKeyDown);
+        return () => {
+            window.removeEventListener('keydown', onKeyDown);
+        }
+    }, []);
+
+    return <StartWorkspaceModalContext.Provider value={{ isStartWorkspaceModalVisible, setIsStartWorkspaceModalVisible }}>
+        {children}
+    </StartWorkspaceModalContext.Provider>;
+}
+
+export const StartWorkspaceModalKeyBinding = `${/(Mac|iPhone|iPod|iPad)/i.test(navigator.platform) ? '⌘' : 'Ctrl+'}O`;
```

Comment:
**suggestion**: Also, Sicne we're here one minor fine tuning we could do is use a better plus sign in terms of alignment, offset, and size. For example, `﹢` instead of `+`.

```suggestion
export const StartWorkspaceModalKeyBinding = `${/(Mac|iPhone|iPod|iPad)/i.test(navigator.platform) ? '⌘﹢' : 'Ctrl﹢'}O`;
```

| BEFORE | AFTER |
|-|-|
| <img width="405" alt="plus-before" src="https://user-images.githubusercontent.com/120486/152975772-2bc81996-0277-46a0-93ff-7912b7482b5c.png"> | <img width="405" alt="plus-after" src="https://user-images.githubusercontent.com/120486/152975777-9ca8b418-20e3-4d0c-a827-d0c77ec4cbb4.png"> |

Reply:
(See [above comment](https://github.com/gitpod-io/gitpod/pull/7715#discussion_r801617528) about OS consistency.)

Reply:
Still, the better plus sign character could be used for Windows systems, right?

Reply:
Ah, I missed the plus sign 😅 honestly here I barely see any difference ("it's just letters" 😝)

Even my browser/system considers these characters equivalent, so it's not easy trying to differentiate them.

I believe VS Code uses a plain regular plus sign in [this file](https://github.com/microsoft/vscode/blob/6dffb2e35788e0814ef074e35df746d644b3603b/src/vs/base/common/keybindingLabels.ts#L68), right?

I'll switch to the new proposed plus sign, but I think this is asking for trouble (e.g. unreadable keybinding on some systems with less advanced unicode support) when we could be using something dead simple. 😋

Reply:
Done 😵